### PR TITLE
feat(batch): support time travel for iceberg source

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Support time travel for iceberg source https://github.com/risingwavelabs/risingwave/issues/15846 https://github.com/risingwavelabs/risingwave/issues/14742 https://github.com/risingwavelabs/risingwave/issues/15822

## Example

```sql
-- FOR SYSTEM_TIME AS OF [STRING | NUMBER] 
-- the timestamp should be a format like [YYYY-MM-DD HH:MM:SS] or an unix timestamp in second
select * from s for system_time as of '2100-01-01 00:00:00+00:00';
select * from s for system_time as of 4102444800;

-- FOR SYSTEM_VERSION AS OF [SNAPSHOT_ID]
select * from s for system_version as of 3023402865675048688;
```

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/source/iceberg/mod.rs [hunk 1]
```diff
@@ -157,16 +157,70 @@ impl SplitEnumerator for IcebergSplitEnumerator {
...
         batch_parallelism: usize,
     ) -> ConnectorResult<Vec<IcebergSplit>> {
         if batch_parallelism == 0 {
             bail!("Batch parallelism is 0. Cannot split the iceberg files.");
         }
         let table = self.config.load_table().await?;
-        let snapshot_id = table.current_table_metadata().current_snapshot_id.unwrap();
+        let snapshot_id = match time_traval_info {
+            Some(IcebergTimeTravelInfo::Version(version)) => {
+                match &table.current_table_metadata().snapshots {
```
Comment (ZENOTME) [inline src/connector/src/source/iceberg/mod.rs:177]: Seems there is an interface to get the specific snapshot version from TableMetadata. https://github.com/icelake-io/icelake/blob/54fd72fbd1dd8c592f05eeeb79223c8a6a33c297/icelake/src/types/in_memory.rs#L2586C12-L2586C20

Comment (liurenjie1024) [reply src/connector/src/source/iceberg/mod.rs:177]: +1

**Review by liurenjie1024 — APPROVED** [review]
Generally LGTM

**Review by fuyufjh — APPROVED** [review]
LGTM

**Pull request merged**